docs/content/docs/test_framework/Mocking.md

```diff
@@ -5,11 +5,73 @@ type: docs
 
 # Mocking in tests
 
+To use a specific state you need to find a suitable `@Rule` or annotation.\
+Thus, test runner knows how to prepare it for the test ([more details]({{< relref "#writing-custom-mocks-for-tests">}})).
+
 ## Mocks (internal)
 
-### Mocking network
+### Networking
+
+#### Mocking a network on http layer
```

Comment:
Does it work without additional logic in http client?
Mock webserver has url. Does url automatically used by applications?

Reply:
1. `AvitoInstrumentTestRunner : InHouseInstrumentationTestRunner()` 
  - Starts a MockWebServer
  - Pass the URL through a Bundle
1. `AvitoTestApp` reads this URL and uses to create Dagger modules

Reply:
Why didn't we write that details? Without them we can confuse our users.

Reply:
This scenario is [tutorial/how-to](https://avito-tech.github.io/avito-android/docs/contributing/docs/#use-appropriate-format-for-the-problem) for internal users. It should explain only how to use it.

I see two other scenarios:

- How to add a new custom rule? 
This explanation is already on this page. Agree, it may be better.
We use the same mechanics for all Rules. There is nothing specific in network.

- How to add ... to a new application?
I think it's better to add it when we explain how to use runner in another project. We may reuse previous scenario.